### PR TITLE
main layout with component navigation

### PR DESCRIPTION
**Pull request by PalmiraBenavente**
- component navigation implementation

- splash screen implementation with VM

-  folders

image
image

-navigation
image

- pixel 3a

image

### CODE REVIEW

File: app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+class WeatherGridFragment : Fragment() {
+
+    private lateinit var viewModel: WeatherGridViewModel
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.weather_grid_fragment, container, false)
```
Comment (BQuinteros) [inline app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt]: Please use binding to inflate this view!

File: app/src/main/res/values/dimen.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="text_size_xbig">50dp</dimen>
+    <dimen name="activity_margin_top_size_big">40dp</dimen>
+    <dimen name="button_text_size_big">25sp</dimen>
+    <dimen name="edit_text_size_big">30sp</dimen>
+    <dimen name="margin_edit_text_normal">20dp</dimen>
+    <dimen name="text_size_normal">20dp</dimen>
```
Comment (mauroLonghi) [inline app/src/main/res/values/dimen.xml]: all text size should be in sp

File: app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.example.weatherandroidapp.R
+import com.example.weatherandroidapp.viewmodels.WeatherGridViewModel
+
+class WeatherGridFragment : Fragment() {
+
+    private lateinit var viewModel: WeatherGridViewModel
```
Comment (BQuinteros) [inline app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt]: Why here you do not use koin to inject the view model as you do it in the other fragment?

File: app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt (hunk 1 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt]: Done

File: app/src/main/res/values/dimen.xml (hunk 2 above)
Comment (PalmiraBenavente) [reply app/src/main/res/values/dimen.xml]: Done

File: app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt (hunk 3 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt]: Done

**Review by BQuinteros — APPROVED** [review]

**Comment by BQuinteros** [comment]
Excellent work <3

File: app/src/main/java/com/example/weatherandroidapp/views/MainActivity.kt [hunk 4]
```diff
@@ -1,7 +1,8 @@
-package com.example.weatherandroidapp
+package com.example.weatherandroidapp.views
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import com.example.weatherandroidapp.R
```
Comment (nicolbidone) [inline app/src/main/java/com/example/weatherandroidapp/views/MainActivity.kt:5]: unused maybe?

File: app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    private val viewModel by viewModel<WeatherGridViewModel>()
+    private lateinit var binding: WeatherGridFragmentBinding
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = WeatherGridFragmentBinding.inflate(layoutInflater)
+        return binding.root
+        TODO("implement view model")
```
Comment (nicolbidone) [inline app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt]: Does this compile?

File: app/src/main/java/com/example/weatherandroidapp/views/WeatherWelcomeFragment.kt [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        binding.buttonMainSelectWeather.setOnClickListener { viewModel.buttonOkPressed() }
+        return binding.root
+    }
+
+    private fun updateUI(navigationEvent: Event<NavigationEvent>?) {
+        when (navigationEvent?.getContentIfNotHandled()) {
+            NavigationEvent.NAVIGATE_TO_WEATHER_DETAILS -> {
+                val action =
+                    WeatherWelcomeFragmentDirections.actionWeatherMainFragmentToWeatherDetailsFragment()
+                view?.findNavController()?.navigate(action)
```
Comment (nicolbidone) [inline app/src/main/java/com/example/weatherandroidapp/views/WeatherWelcomeFragment.kt]: please avoid single-use vars

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -4,15 +4,17 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        app:defaultNavHost="true"
         app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
-
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
```
Comment (nicolbidone) [inline app/src/main/res/layout/activity_main.xml]: if width = match_parent , constraints to start & end are useless

File: app/src/main/res/layout/activity_main.xml [hunk 8]
```diff
@@ -4,15 +4,17 @@
...
+        app:layout_constraintTop_toTopOf="parent"
```
Comment (nicolbidone) [inline app/src/main/res/layout/activity_main.xml]: if width = match_parent , constraints to top & bottom are useless

File: app/src/main/res/navigation/nav_graph.xml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+            android:id="@+id/action_weatherMainFragment_to_weatherDetailsFragment"
+            app:destination="@id/weatherDetailsFragment" />
+    </fragment>
+
+    <fragment
+        android:id="@+id/weatherDetailsFragment"
+        android:name="com.example.weatherandroidapp.views.WeatherGridFragment"
+        android:label="weather_details_fragment"
+        tools:layout="@layout/weather_grid_fragment" />
+</navigation>
```
Comment (nicolbidone) [inline app/src/main/res/navigation/nav_graph.xml:23]: ❤️

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -1,3 +1,8 @@
 <resources>
     <string name="app_name">WeatherAndroidApp</string>
+    <string name="text_main_activity_weather">WELCOME!</string>
+    <string name="button_main_ok">OK</string>
+    <string name="text_edit_country">Enter your city</string>
+    <string name="text_main_activity_subtitle_weather">Do you want to know the climate of Tandil?</string>
+    <string name="text_weather_grid">Weather Grid!</string>
```
Comment (nicolbidone) [inline app/src/main/res/values/strings.xml:7]: climate or weather?

File: app/src/main/java/com/example/weatherandroidapp/viewmodels/WeatherGridViewModel.kt [hunk 11]
```diff
@@ -0,0 +1,7 @@
+package com.example.weatherandroidapp.viewmodels
+
+import androidx.lifecycle.ViewModel
+
+class WeatherGridViewModel : ViewModel() {
+    // TODO: Implement the ViewModel
+}
```
Comment (nicolbidone) [inline app/src/main/java/com/example/weatherandroidapp/viewmodels/WeatherGridViewModel.kt:7]: is this neccesary?

File: app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt (hunk 5 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt]: emm, yes yes

File: app/src/main/java/com/example/weatherandroidapp/views/WeatherWelcomeFragment.kt (hunk 6 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/weatherandroidapp/views/WeatherWelcomeFragment.kt]: Done

File: app/src/main/res/layout/activity_main.xml (hunk 7 above)
Comment (PalmiraBenavente) [reply app/src/main/res/layout/activity_main.xml]: Done

File: app/src/main/res/layout/activity_main.xml (hunk 8 above)
Comment (PalmiraBenavente) [reply app/src/main/res/layout/activity_main.xml]: Done

File: app/src/main/res/values/strings.xml (hunk 10 above)
Comment (PalmiraBenavente) [reply app/src/main/res/values/strings.xml:7]: weather weather jaja

File: app/src/main/java/com/example/weatherandroidapp/utils/WeatherApp.kt [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    override fun onCreate() {
+        super.onCreate()
+        startKoin {
+            androidContext(this@WeatherApp)
+            androidLogger(Level.DEBUG)
+            modules(listOf(viewModelsModule))
+        }
+    }
+
+    override fun attachBaseContext(base: Context?) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/weatherandroidapp/utils/WeatherApp.kt]: as you are only calling the super, this is unnecessary, please remove this method

File: app/src/main/java/com/example/weatherandroidapp/utils/WeatherApp.kt [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+import com.example.weatherandroidapp.di.viewModelsModule
+import org.koin.android.ext.koin.androidContext
+import org.koin.android.ext.koin.androidLogger
+import org.koin.core.context.startKoin
+import org.koin.core.logger.Level
+
+class WeatherApp : Application() {
+    override fun onCreate() {
+        super.onCreate()
+        startKoin {
```
Comment (SartiMau) [inline app/src/main/java/com/example/weatherandroidapp/utils/WeatherApp.kt:13]: excellent!

File: app/src/main/java/com/example/weatherandroidapp/viewmodels/WeatherGridViewModel.kt (hunk 11 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/weatherandroidapp/viewmodels/WeatherGridViewModel.kt:7]: No, but I had created the empty weather grid fragment for the interaction, it seemed good to me that it has the respective view model, and I will use it in the future

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/weatherandroidapp/views/MainActivity.kt (hunk 4 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/weatherandroidapp/views/MainActivity.kt:5]: use to associated the activity to a xml

File: app/src/main/java/com/example/weatherandroidapp/utils/WeatherApp.kt (hunk 12 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/weatherandroidapp/utils/WeatherApp.kt]: Done

File: app/src/main/java/com/example/weatherandroidapp/views/MainActivity.kt (hunk 4 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/weatherandroidapp/views/MainActivity.kt:5]: it's not necessary in this case?

**Review by SartiMau — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**